src/precice/impl/Participant.hpp

```diff
@@ -21,7 +21,7 @@
 
 namespace precice {
 namespace impl {
-struct DataContext;
+class DataContext;
```

Comment:
Do we still need / want this forward declaration?

Reply:
I replaced the forward declaration with a normal `#include "precice/impl/DataContext.hpp"` in 5dae15a914526ff92a84bed194d278cbf2f71af3.